src/main/java/org/jenkinsci/plugins/workflow/libs/SCMSourceRetriever.java

```diff
@@ -90,6 +91,37 @@ public SCMSource getScm() {
         retrieve(name, version, true, target, run, listener);
     }
 
+    private static <T> T retrySCMOperation(TaskListener listener, Callable<T> task) throws Exception{
+        T ret = null;
+        for (int retryCount = Jenkins.get().getScmCheckoutRetryCount(); retryCount >= 0; retryCount--) {
+            try {
+                ret = task.call();
+                break;
+            }
+            catch (AbortException e) {
+                // abort exception might have a null message.
+                // If so, just skip echoing it.
+                if (e.getMessage() != null) {
+                    listener.error(e.getMessage());
+                }
+            }
+            catch (InterruptedIOException e) {
+                throw e;
+            }
+            catch (IOException e) {
+                // checkout error not yet reported
+                listener.error("Checkout failed").println(Functions.printThrowable(e).trim()); // TODO 2.43+ use Functions.printStackTrace
```

Comment:
nit: I know you're just moving existing code around, but the baseline is high enough to clean up this todo:
```suggestion
                Functions.printStackTrace(e, listener.error("Checkout failed");
```